components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -692,27 +702,41 @@ public void validateConfirmationCode(String code, String recoveryStep) throws Id
     }
 
     private void auditPasswordRecovery(String action, String notificationChannel, User user, String errorMsg,
-                                       String result) {
+                                       String result, UserRecoveryData userRecoveryData,
+                                       String notificationTemplateType) {
 
         JSONObject dataObject = new JSONObject();
         dataObject.put(AuditConstants.REMOTE_ADDRESS_KEY, MDC.get(AuditConstants.REMOTE_ADDRESS_QUERY_KEY));
         dataObject.put(AuditConstants.USER_AGENT_KEY, MDC.get(AuditConstants.USER_AGENT_QUERY_KEY));
         dataObject.put(AuditConstants.NOTIFICATION_CHANNEL, notificationChannel);
         dataObject.put(AuditConstants.SERVICE_PROVIDER_KEY, MDC.get(AuditConstants.SERVICE_PROVIDER_QUERY_KEY));
         dataObject.put(AuditConstants.USER_STORE_DOMAIN, user.getUserStoreDomain());
+        if (userRecoveryData != null) {
+            dataObject.put(AuditConstants.RECOVERY_SCENARIO, userRecoveryData.getRecoveryScenario().name());
+            dataObject.put(AuditConstants.RECOVERY_STEP, userRecoveryData.getRecoveryStep().name());
+        }
+        dataObject.put(AuditConstants.TENANT_DOMAIN, user.getTenantDomain());
+        dataObject.put(AuditConstants.NOTIFICATION_TEMPLATE_TYPE, notificationTemplateType);
+
         if (AUDIT_FAILED.equals(result)) {
             dataObject.put(AuditConstants.ERROR_MESSAGE_KEY, errorMsg);
         }
         Utils.createAuditMessage(action, user.getUserName(), dataObject, result);
     }
 
-    private void auditPasswordReset(String action, User user, String errorMsg, String result) {
+    private void auditPasswordReset(String action, User user, String errorMsg, String result,
+                                    UserRecoveryData userRecoveryData) {
```

Comment:
Can't we just pass the scenario and step as String values? We are getting the user from the userRecoveryData, right? So in that case passing the user separately and again passing the recovery data object is not nice. WDYT?

Reply:
Those 2 are not strings. They are enums. But as I understand the enums are defined in a wrong way in UserRecoveryData class. Therefore, if we change this as you said, it looks like this. 
```
private void auditPasswordReset(String action, User user, String errorMsg, String result,
                                    Enum recoveryScenario, Enum recoveryStep) {
```

Is that ok?

Reply:
I think we can pass the String value of that enum. Can't we?